python/FindPythonInterp.cmake

```diff
@@ -0,0 +1,180 @@
+                            "import sys; sys.stdout.write(';'.join([str(x) for x in sys.version_info[:3]]))"
+                    OUTPUT_VARIABLE _VERSION
+                    RESULT_VARIABLE _PYTHON_VERSION_RESULT
+                    ERROR_QUIET)
+    if(NOT _PYTHON_VERSION_RESULT)
+        string(REPLACE ";" "." PYTHON_VERSION_STRING "${_VERSION}")
+        list(GET _VERSION 0 PYTHON_VERSION_MAJOR)
+        list(GET _VERSION 1 PYTHON_VERSION_MINOR)
+        list(GET _VERSION 2 PYTHON_VERSION_PATCH)
+        if(PYTHON_VERSION_PATCH EQUAL 0)
+            # it's called "Python 2.7", not "2.7.0"
+            string(REGEX REPLACE "\\.0$" "" PYTHON_VERSION_STRING "${PYTHON_VERSION_STRING}")
+        endif()
+    else()
+        # sys.version predates sys.version_info, so use that
+        execute_process(COMMAND "${PYTHON_EXECUTABLE}" -c "import sys; sys.stdout.write(sys.version)"
+                        OUTPUT_VARIABLE _VERSION
+                        RESULT_VARIABLE _PYTHON_VERSION_RESULT
+                        ERROR_QUIET)
+        if(NOT _PYTHON_VERSION_RESULT)
+            string(REGEX REPLACE " .*" "" PYTHON_VERSION_STRING "${_VERSION}")
+            string(REGEX REPLACE "^([0-9]+)\\.[0-9]+.*" "\\1" PYTHON_VERSION_MAJOR "${PYTHON_VERSION_STRING}")
+            string(REGEX REPLACE "^[0-9]+\\.([0-9])+.*" "\\1" PYTHON_VERSION_MINOR "${PYTHON_VERSION_STRING}")
+            if(PYTHON_VERSION_STRING MATCHES "^[0-9]+\\.[0-9]+\\.([0-9]+)")
+                set(PYTHON_VERSION_PATCH "${CMAKE_MATCH_1}")
+            else()
+                set(PYTHON_VERSION_PATCH "0")
+            endif()
+        else()
+            # sys.version was first documented for Python 1.5, so assume
+            # this is older.
+            set(PYTHON_VERSION_STRING "1.4")
+            set(PYTHON_VERSION_MAJOR "1")
+            set(PYTHON_VERSION_MINOR "4")
+            set(PYTHON_VERSION_PATCH "0")
+        endif()
+    endif()
+    unset(_PYTHON_VERSION_RESULT)
+    unset(_VERSION)
+endif()
+
+# This is a hack to make FindPackageHandleStandardArgs not failing
+# when EXACT version of the form "3" or "3.2" is required and
+# PYTHONLIBS_VERSION_STRING is "3.2.4".
+# This hack is in CMake 3.2 and above.
+set(_PYTHON_VERSION_STRING "${PYTHON_VERSION_STRING}")
+if(PythonInterp_FIND_VERSION)
+    set(PYTHON_VERSION_STRING "${PythonInterp_FIND_VERSION_MAJOR}")
+    if(PythonInterp_FIND_VERSION_COUNT GREATER 1)
+        set(PYTHON_VERSION_STRING "${PYTHON_VERSION_STRING}.${PythonInterp_FIND_VERSION_MINOR}")
+    endif(PythonInterp_FIND_VERSION_COUNT GREATER 1)
+    if(PythonInterp_FIND_VERSION_COUNT GREATER 2)
+        set(PYTHON_VERSION_STRING "${PYTHON_VERSION_STRING}.${PythonInterp_FIND_VERSION_TWEAK}")
+    endif(PythonInterp_FIND_VERSION_COUNT GREATER 2)
+endif()
+
+# handle the QUIETLY and REQUIRED arguments and set PYTHONINTERP_FOUND to TRUE if
+# all listed variables are TRUE
+include(FindPackageHandleStandardArgs)
```

Comment:
This line was originally
```cmake
include(${CMAKE_CURRENT_LIST_DIR}/FindPackageHandleStandardArgs.cmake)
```
I'm not sure of why they including it this way. I guess it was to make sure `include` finds the good version.